src/embeddings.py

```
@@ -0,0 +1,224 @@
+        List of (chunk_text, token_count) tuples
+    """
+    if len(encoded_tokens) <= max_tokens:
+        return [(text, len(encoded_tokens))]
+
+    # Use 20% overlap for better semantic continuity
+    overlap_tokens = int(max_tokens * 0.2)
+    step_size = max_tokens - overlap_tokens
+
+    return [
+        (
+            encoding.decode(encoded_tokens[i : i + max_tokens]),
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)